torch/testing/_internal/distributed/_sharded_tensor/__init__.py

```diff
@@ -0,0 +1,85 @@
+import sys
+from functools import partial, wraps
+
+import torch
+import torch.distributed as dist
+from torch.distributed import rpc
+from torch.testing._internal.common_distributed import (
+    TEST_SKIPS,
+)
+
+
+class ShardedTensorTestBase(object):
+    @property
+    def world_size(self):
+        return 4
+
+    def init_pg(self):
+        dist.init_process_group(
+            backend="nccl",
+            world_size=self.world_size,
+            rank=self.rank,
+            init_method=f"file://{self.file_name}",
+        )
+
+    def init_rpc(self):
+        rpc_backend_options = rpc.TensorPipeRpcBackendOptions()
+        rpc_backend_options.init_method = f"file://{self.file_name}"
+        for rank in range(self.world_size):
+            rpc_backend_options.set_device_map(
+                f"worker{rank}", {rank: self.rank, self.rank: rank}
+            )
```

Comment:
maybe make init_pg and init_rpc as internal method since you already have init_comms below?


_init_pg
_init_rpc

Reply:
There are certain tests that call `init_pg` and `init_rpc` directly instead of going through `init_comms`.